emukit/core/optimization/anchor_points_generator.py

```diff
@@ -76,9 +76,36 @@ def __init__(self, space: ParameterSpace, acquisition: Acquisition, num_samples=
         self.acquisition = acquisition
         super(ObjectiveAnchorPointsGenerator, self).__init__(space, num_samples)
 
-    def get_anchor_point_scores(self, X) -> np.array:
+    def get_anchor_point_scores(self, X: np.ndarray) -> np.ndarray:
         """
         :param X: The samples at which to evaluate the criterion
         :return:
         """
         return self.acquisition.evaluate(X).flatten()
+
+
+class ConstrainedObjectiveAnchorPointsGenerator(AnchorPointsGenerator):
```

Comment:
Maybe we don't need a new class here? ObjectiveAnchorPointsGenerator is just ConstrainedObjectiveAnchorPointsGenerator with empty list of constraints, isn't it?

Reply:
Good point, I'll change this